detecting_metamask.md

```diff
@@ -2,65 +2,64 @@
 
 MetaMask enables your website to read from the blockchain and propose transactions to the current user. To enjoy these benefits, you have to first check if the user has an Ethereum-compatible browser like MetaMask.
 
-## Detecting MetaMask
-
-Metamask/Mist currently inject their interface into page as global `web3` object.
+## What is an Ethereum provider?
 
-```javascript
-window.addEventListener('load', function() {
+A provider is a low-level object with one primary method: `sendAsync(options, callback)`. The `options` object has several useful fields: `method` (the method name), `from` (the sender address), `params` (an array of parameters to send to that method).
 
-  // Checking if Web3 has been injected by the browser (Mist/MetaMask)
-  if (typeof web3 !== 'undefined') {
+These methods all correspond directly to the options in [the RPC provider spec](https://github.com/ethereum/wiki/wiki/JSON-RPC).
 
-    // Use the browser's ethereum provider
-    var provider = web3.currentProvider
+## Detecting MetaMask
 
-  } else {
-    console.log('No web3? You should consider trying MetaMask!')
-  }
+MetaMask and other privacy-focused browsers require dapps to request a provider before using it. Some older browsers may still inject a Web3 instance automatically, so it's best practice to handle both cases:
 
-})
+```javascript
+// If web3 is not injected (modern browsers)...
```

Comment:
What about using the code provided in the [medium post announcing the changes](https://medium.com/metamask/https-medium-com-metamask-breaking-change-injecting-web3-7722797916a8) instead? That one uses the `enable()` API, which is the standard way of requesting access.
See #119 for a PR that uses that code snippet.